core/src/main/java/org/elasticsearch/action/search/SearchResponse.java

```diff
@@ -240,4 +330,87 @@ public void writeTo(StreamOutput out) throws IOException {
     public String toString() {
         return Strings.toString(this);
     }
+
+    public static class InternalSearchResponse {
```

Comment:
I just checked where InternalSearchResponse is used, all places that I found where this is instantiated it is later wrapped in a SearchResponse in the same method, so I think it's never sent across the wire independently.